benchmarks/benchmark-fdot.C

```diff
@@ -0,0 +1,127 @@
+/* -*- mode: C++; tab-width: 4; indent-tabs-mode: t; c-basic-offset: 4 -*- */
+// vim:sts=4:sw=4:ts=4:noet:sr:cino=>s,f0,{0,g0,(0,\:0,t0,+0,=s
+
+
+/* Copyright (c) FFLAS-FFPACK
+ * Written by Philippe LEDENT <philippe.ledent@etu.univ-grenoble-alpes.fr>
+ * ========LICENCE========
+ * This file is part of the library FFLAS-FFPACK.
+ *
+ * FFLAS-FFPACK is free software: you can redistribute it and/or modify
+ * it under the terms of the  GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * ========LICENCE========
+ */
+
+#include "fflas-ffpack/fflas-ffpack-config.h"
+#include <iostream>
+#include <givaro/modular.h>
+
+#include "fflas-ffpack/fflas-ffpack.h"
+#include "fflas-ffpack/utils/timer.h"
+#include "fflas-ffpack/utils/fflas_io.h"
+#include "fflas-ffpack/paladin/parallel.h"
+#include "fflas-ffpack/paladin/fflas_plevel1.h"
+#include "fflas-ffpack/utils/args-parser.h"
+
+
+using namespace std;
+using namespace FFLAS;
+using namespace FFPACK;
+int main(int argc, char** argv) {
+
+	size_t iter      = 3;
+	size_t N      = 5000;
+ 	size_t BS      = 5000;
+//     size_t q = 101;
+	size_t p=0;
+	std::string file = "";
+  
+	Argument as[] = {
+		{ 'n', "-n N", "Set the dimension of the matrix C.",TYPE_INT , &N },
+// 		{ 'q', "-q Q", "Set the field characteristic (-1 for random).",         TYPE_INT , &q },
+		{ 'i', "-i R", "Set number of repetitions.",		TYPE_INT , &iter },
+		{ 's', "-i S", "Size of the integers.",				TYPE_INT , &BS },
+		{ 'p', "-p P", "0 for sequential, 1 for parallel.",	TYPE_INT , &p },
+		END_OF_ARGUMENTS
+	};
+  
+	FFLAS::parseArguments(argc,argv,as);
+  
+    typedef Givaro::ZRing<Givaro::Integer> Field;
+    Field F;
+    Givaro::GivRandom generator;
+    Givaro::IntegerDom IPD;
+
+// 	typedef Givaro::ModularBalanced<double> Field;
+// 	Field F(q);
+	
+	Field::Element_ptr A, B;
+	Field::Element d; F.init(d);
+  
+	Givaro::OMPTimer chrono, time; time.clear();
+  
+	for (size_t i=0;i<iter;++i){
+		A = fflas_new(F, N);
+		B = fflas_new(F, N);
+        PARFOR1D(j,N,SPLITTER(), {
```

Comment:
Why not use the simpler `pfrand`  of `RandomMatrix` initializer of FFLAS ? See `benchmark-charpoly.C` or `benchmark-fgemm.C`.